src/ResourceManager/Compute/Commands.Compute/Strategies/ComputeRp/VirtualMachineStrategy.cs

```diff
@@ -112,7 +112,8 @@ static class VirtualMachineStrategy
                             CreateOption = DiskCreateOptionTypes.Attach,
                             OsType = osType,
                             ManagedDisk = engine.GetReference(disk, StorageAccountTypes.PremiumLRS),
-                        }
+                        },
+                        DataDisks = DataDiskStrategy.CreateDataDisks(0, dataDisks)
```

Comment:
Why is the data disk count always 0?

Reply:
This configuration for VHD OS disks only. I have no idea how to extract this information from VHD. I'm not sure if it's possible.